### PR TITLE
Remove binding for CircuitBreakerRegistry

### PR DESCRIPTION
**Pull request by tobias-bahls**
This removes the explicit binding of a `CircuitBreakerRegistry`. Users of the library will have to bind their own `CircuitBreakerRegistry`.

### CODE REVIEW

**Comment by cberg-zalando** [comment]
Then please add this information to the README.md. :stuck_out_tongue:

File: README.md [hunk 1]
```diff
@@ -28,6 +28,9 @@ The LogbookHandler expects an obfuscation function of type `Function<String, Str
 to be injected with the name `logbook.pathObfuscator`. This function prevents confidential path parameters in the URI to be 
 logged as plain text.
 
+###  Usage of OAuth2-related functionality
+In order to safely call OAuth2 providers, this library makes use of the Circuit Breaker library `javaslang-circuitbreaker`. Users
+of this library are expected to bind an instance to `CircuitBreakerRegistry`, which will be used to instantiate circuit breakers.
```
Comment (cberg-zalando) [inline README.md]: What about adding a link to the `javaslang-circuitbreaker` GHE page?

**Comment by cberg-zalando** [comment]
:+1:

**Comment by fatroom** [comment]
👍

**Pull request merged**